src/main/resources/native/android/dalvik/source/MainActivity.java

```diff
@@ -230,39 +257,51 @@ public boolean dispatchKeyEvent(final KeyEvent event) {
     protected void onDestroy() {
         Log.v(TAG, "onDestroy");
         super.onDestroy();
+        notifyLifecycleEvent("destroy");
         android.os.Process.killProcess(android.os.Process.myPid());
     }
 
     @Override
     protected void onPause() {
         Log.v(TAG, "onPause");
         super.onPause();
+        notifyLifecycleEvent("pause");
     }
 
     @Override
     protected void onResume() {
         Log.v(TAG, "onResume");
         super.onResume();
+        notifyLifecycleEvent("resume");
         Log.v(TAG, "onResume done");
     }
 
     @Override
     protected void onStart() {
         Log.v(TAG, "onStart");
         super.onStart();
+        notifyLifecycleEvent("start");
         Log.v(TAG, "onStart done");
     }
 
     @Override
     protected void onRestart() {
         Log.v(TAG, "onRestart");
         super.onRestart();
+        notifyLifecycleEvent("restart");
     }
 
     @Override
     protected void onStop() {
         Log.v(TAG, "onStop");
         super.onStop();
+        notifyLifecycleEvent("stop");
+    }
+
+    private void notifyLifecycleEvent(String event) {
+        if (graalStarted) {
+            nativeDispatchLifecycleEvent(event, event.length());
```

Comment:
I was wondering if passing an int instead of a string might be easier? The int values can be easily mapped to the matching LifecycleEvent enum in Attach?

Reply:
So far we only listen for pause/resume lifecycle events, but there are more (start, stop, ...). Adding an int could cause issues if we later on add any of those events.